framework/include/loops/ComputeFVFluxThread.h

```diff
@@ -126,24 +126,47 @@ class ThreadedFaceLoop
 
   /// The subdomain for the last neighbor
   SubdomainID _old_neighbor_subdomain;
+
+  /// Holds caught runtime error messages
+  std::string _error_message;
+
+private:
+  /// Whether this is the zeroth threaded copy of this body
+  const bool _zeroth_copy;
 };
 
 template <typename RangeType>
 ThreadedFaceLoop<RangeType>::ThreadedFaceLoop(FEProblemBase & fe_problem,
                                               const std::set<TagID> & tags)
-  : _fe_problem(fe_problem), _mesh(fe_problem.mesh()), _tags(tags)
+  : _fe_problem(fe_problem), _mesh(fe_problem.mesh()), _tags(tags), _zeroth_copy(true)
 {
+  Moose::_throw_on_error = true;
 }
 
 template <typename RangeType>
 ThreadedFaceLoop<RangeType>::ThreadedFaceLoop(ThreadedFaceLoop & x, Threads::split /*split*/)
-  : _fe_problem(x._fe_problem), _mesh(x._mesh), _tags(x._tags)
+  : _fe_problem(x._fe_problem), _mesh(x._mesh), _tags(x._tags), _zeroth_copy(false)
 {
 }
 
 template <typename RangeType>
 ThreadedFaceLoop<RangeType>::~ThreadedFaceLoop()
 {
+  if (_zeroth_copy)
+  {
+    Moose::_throw_on_error = false;
```

Comment:
What do you mean? The default is to abort but that is dangerous in threaded regions because when you error you write to console for which you have to lock a mutex and then an aborting thread will attempt to destroy the mutex… while another thread may have locked the mutex for its own console output. This is the reason we kept seeing failures on @grmnptr’s error test with threads because of these races on the mutex due to locking and destruction. By throwing instead we can gracefully catch the exception and wait to abort until we’ve exited the threaded region

Reply:
Setting throw on error to false is equivalent to saying “abort on error”

Reply:
this is a global variable isnt it?
once you modified it and you are done with it you could reset it to its previous status, so that (an hypothetical) someone who relied on it being true for testing does not get surprised

Reply:
Ok so you think people will be using ThreadedFaceLoops in unit testing. That is the only time where the incoming value would be `true` instead of `false`.

Reply:
Updated to record the incoming value of `Moose::_throw_on_error` and restore it upon destruction